tests/math/test_expectation_value.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit tests for differentiable matrix-vector expectation values.
+"""
+
+import numpy as onp
+import pytest
+
+import pennylane as qml
+from pennylane import numpy as np
+
+pytestmark = pytest.mark.all_interfaces
+
+tf = pytest.importorskip("tensorflow", minversion="2.1")
+torch = pytest.importorskip("torch")
+jax = pytest.importorskip("jax")
+jnp = pytest.importorskip("jax.numpy")
+
+
+class TestExpectationValueMath:
+    """Tests for Expectation value of a operator for a state vector."""
+
+    ops_vs_vecstates = [
+        ([[1, 0], [0, 0]], [1, 0], 1),
```

Comment:
Can we also include a couple of tests for state vectors where N=2? 

Possibly also a test akin to the example above that uses a matrix from a PL operator.

Reply:
Added! I have a question though about the case of batched elements. Suppose a batch $X$ containing few vector states $x_i$, they are all supposed to have the same dimension, let's say $N$? Or is it allowed to mix it?

Reply:
sadly, no mixing - a batched set of states must all have the same dimension, namely because numerical packages (numpy) don't support ragged arrays